src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -571,11 +570,13 @@ protected internal override Task OnCloseAsync(TimeSpan timeout)
         protected override void OnClosed()
         {
             base.OnClosed();
-            var httpClientToDispose = _httpClient;
-            if (httpClientToDispose != null)
+            if (_httpClientCache != null && !_httpClientCache.IsDisposed)
```

Comment:
`_httpClientCache` could become null after the null check and right before the check on `_httpClientCache.IsDisposed` . I think we can use `Interlocked.CompareExchange` here to get the instance and set `_httpClientCache` to null. Then we can also omit the lock below.